recipes/newrelic/apm/php-agent/php.yml

```diff
@@ -29,20 +29,80 @@ inputVars:
 install:
   version: "3"
 
+  vars:
+    TMP_INSTALL_DIR:
+      sh: mktemp -d /tmp/newrelic-php-agent.XXXXXX
+
   env:
     NEW_RELIC_DISTRIBUTED_TRACING_ENABLED: true
 
   tasks:
     default:
       cmds:
-        - task: install_packages
         - task: create_shared
+        - task: detect_services
+        - task: install_packages
         - task: config_newrelic_ini
         - task: finalize_agent_config
         - task: restart_services
         - task: send_transaction
         - task: delete_shared
 
+    create_shared:
+      cmds:
+        - rm -rf {{.TMP_INSTALL_DIR}}
+        - mkdir -p {{.TMP_INSTALL_DIR}}
+        - echo "Created temporary directory {{.TMP_INSTALL_DIR}} for installation."
+
+    detect_services:
+      cmds:
+        - echo "Detect services."
+        - |
+          cd {{.TMP_INSTALL_DIR}}
+          rm -f {{.TMP_INSTALL_DIR}}/nonpriv_users.txt
+          rm -f {{.TMP_INSTALL_DIR}}/processes_to_restart.txt
+
+          #
+          # For guided installs, we currently support `php-fpm`, `apache`, and `nginx`.
```

Comment:
I can take it out, added it for completeness as it doesn't do anything if it's not detected, but when we start with nginx, we are ready for it.

Reply:
If we are restarting all the processes we find, leaving it in may actually increase install success rates as this is the key (only?) thing needed to handle non-fpm nginx scenarios.

Reply:
But wait, why don't we support nginx with php-fpm?
I thought that was the [MMF](https://github.com/newrelic/newrelic-php-agent/issues/93)?
Based on the text from the MMF, I thought nginx was the focus:
```
For users of Debian or RHEL derived systems, running PHP under apache or fpm/nginx, the New Relic Open Installer library can be used to easily install and configure the PHP Agent on their systems.

This epic targets the initial buildout of supporting this, with a focus on FPM+nginx under Debian. Additional target environments will be supported in the subsequent epic.
```
Is this incorrect?

Reply:
Either way, this recipe handles the combo of fpm/apache/nginx.

Reply:
We do support that case. When FPM is present, restarting it is sufficient to load the agent. nginx doesn't need to be restarted in that case. If there is a way to run a PHP site with nginx without FPM, restarting nginx is likely to be needed.

Reply:
In that case, I think we only need to restart the first process we detect.
We detect in this order: fpm, apache, nginx
If we detect fpm, we restart that and move on even if we also detect apache and/or nginx.
Otherwise, if fpm doesn't exist we will detect apache or nginx and restart that.